client/src/pages/Profile/ProfileAvailability/ProfileAvailability.tsx

```diff
@@ -0,0 +1,160 @@
+        your availability
+      </Typography>
+      <Box display="inline" className={classes.availability}>
+        <Icon color="secondary">
+          <DateRangeIcon />
+        </Icon>
+        <Button onClick={() => setHidden(!hidden)} className={classes.dateRange}>{`
+              ${format(from, 'dd-')}${`${format(to, 'dd MMMM yyyy')}`}`}</Button>
+      </Box>
+      {!hidden && (
+        <Box display="inline" className={classes.datePicker}>
+          <DateRange
+            startDatePlaceholder="Start Date"
+            endDatePlaceholder="End Date"
+            onChange={(ranges) => setSelection(ranges.rollup)}
+            minDate={new Date()}
+            ranges={[selection]}
+          />
+          <Button variant="contained" size="small" startIcon={<SaveIcon />} onClick={() => setHidden(!hidden)}>
+            Save
+          </Button>
+        </Box>
+      )}
+      <Card variant="outlined">
+        {availability &&
+          intervalOfDays?.map((day, index) => (
+            <>
+              <Box className={classes.content} key={`${day} ${index}`}>
+                <Switch
+                  checked={getChecked(day)}
+                  onChange={() => handleAvailibility(day)}
+                  name="availability"
+                  inputProps={{ 'aria-label': 'secondary checkbox' }}
+                />
+                <Typography variant="h6" className={classes.dayText}>
+                  {`${getDate(day)} ${MONTHS[getMonth(day)]},`}
+                  <Typography color="textSecondary" display="inline">
+                    {DAYS[getDay(day)]}
+                  </Typography>
+                </Typography>
+                <CardContent className={classes.time}>
+                  <Typography className={classes.text}>From</Typography>
+                  <TextField
+                    id="time"
+                    type="time"
+                    defaultValue="09:00"
+                    InputLabelProps={{
+                      shrink: true,
+                    }}
+                    inputProps={{
+                      step: 300, // 5 min
+                    }}
+                    variant="outlined"
+                    className={classes.textField}
+                  />
+                  <Typography className={classes.text}>To</Typography>
+                  <TextField
+                    id="time"
+                    type="time"
+                    defaultValue="18:00"
+                    InputLabelProps={{
+                      shrink: true,
+                    }}
+                    inputProps={{
+                      step: 300, // 5 min
+                    }}
+                    variant="outlined"
+                    className={classes.textField}
+                  />
+                </CardContent>
+              </Box>
+              <Divider />
+            </>
+          ))}
+      </Card>
+    </Grid>
+  );
+}
+
+export default ProfileAvailability;
```

Comment:
This component looks very good overall - the UI/UX seems on point imho! 🎉 

I want to take this opportunity to mention something that I believe is going to be very helpful in the long run - I think we can clean up our code significantly if we avoid inline/anonymous definitions. Material UI is very good at this, take a look at the implementation of the `TextField` component [here](https://github.com/mui-org/material-ui/blob/v4.x/packages/material-ui/src/TextField/TextField.js): The UI of the component (e.g. what they return) is very easy to understand and follow because they handle all the logic outside of the "rendering phase" - the rest is simple JSX. 

No need to change anything specific here, rather I'd love to see more of this idea in your future PRs ❤️